macros/contexts/contextSignificantFigures.pl

```diff
@@ -84,32 +83,28 @@ sub new {
 	my $self    = shift;
 	my $class   = ref($self) || $self;
 	my $context = (Value::isContext($_[0]) ? shift : $self->context);
-	my ($value, %opts) = @_;
+	my $value   = shift;
+	my %opts    = @_;
+	my $n       = $opts{sigfigs};
 
-	# If a SignificantFigure::Number is passed in, make a copy.
 	if (Value::isValue($value) && $value->{sigfigs}) {
 		my $copy = $value->copy->inContext($context);
 		$copy->sigfigs($n) if defined($n) && $n != $copy->N;
 		return $copy;
 	}
-	# This handles formulas.
 	if (!Value::matchNumber($value)) {
 		$value = Value::makeValue($value, context => $context);
 		return $value if Value::isFormula($value);
 		Value::Error("Can't convert %s to %s", Value::showClass($value), Value::showClass($self));
 	}
 	return $value->eval if Value::isFormula($value);
-	my $n = $opts{sigfigs};
 
 	if (!defined $n) {
 		my $digits = $value;
-		if ($value == 0) {    # 0 is has infinite sig figs
-			$n = 'inf';
-		} elsif ($value !~ m/\./) {    # it is an integer
-			$digits =~ s/^[+-]?0*//;    # remove signs and leading zeros;
-			$digits =~ s/0+$//;         # remove trailing zeros;
+		if ($value !~ m/[.Ee]/) {
+			$digits =~ s/0+$//;
 		} else {
-			$digits =~ s/[Ee].*$//;     # remove exponent, if any
+			$digits =~ s/[Ee].*$//;    # remove exponent, if any
 			if ($value == 0) {
 				$digits =~ s/^[+-]?0+(.(?:\.|$))/$1/;    # remove leading 0s before the decimal
```

Comment:
I realized that there needs to be one more line after this one:

``` perl
				$digits =~ s/^\./0./g;    # add initial zero, if needed
```

so that something like `Compute(".00")` will end up being `0.00` not `0.0` (two digits).

Reply:
I'm testing 0.00, which should have 2 sig figs.  Is line 109 supposed to replace this will .00 ?  It isn't doing that. 

And then why do we want to add another leading 0?  That ends up have an extra sig fig, doesn't it?

Reply:
I'm thinking this this block is only for numbers that are equal to 0, then 1) if the result is 0, it should be 0 with infinite sigfigs or 2) has the form 0.000  and in the 2nd case would `s/^[+-]?0+\.(0+)$/$1/` give the correct number of  sigfigs?

Reply:
> in the 2nd case would `s/^[+-]?0+\.(0+)$/$1/` give the correct number of sigfigs?

No, this will be one short, in general, and since it requires there to be a leading 0, it won't handle things like `+.00`.